pkg/subctl/cmd/diagnose/globalnet.go

```diff
@@ -0,0 +1,162 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package diagnose
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd"
+	"github.com/submariner-io/submariner/pkg/globalnet/constants"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcsClientset "sigs.k8s.io/mcs-api/pkg/client/clientset/versioned"
+)
+
+func init() {
+	diagnoseCmd.AddCommand(&cobra.Command{
+		Use:   "globalnet",
+		Short: "Check globalnet configuration",
+		Long:  "This command checks globalnet configuration",
+		Run: func(command *cobra.Command, args []string) {
+			cmd.ExecuteMultiCluster(restConfigProducer, checkGlobalNet)
+		},
+	})
+}
+
+func checkGlobalNet(cluster *cmd.Cluster) bool {
+	status := cli.NewStatus()
+
+	if cluster.Submariner == nil {
+		status.Start(cmd.SubmMissingMessage)
+		status.EndWith(cli.Warning)
+
+		return true
+	}
+
+	status.Start("Checking Globalnet configuration")
+
+	if cluster.Submariner.Spec.GlobalCIDR != "" {
+		checkClusterGlobalEgressIps(cluster, status)
+		checkGlobalEgressIps(cluster, status)
+		checkGlobalIngressIps(cluster, status)
+
+		if status.HasFailureMessages() {
+			status.EndWith(cli.Failure)
+			return false
+		}
+
+		status.EndWithSuccess("Globalnet is enabled and properly configured")
+	} else {
+		status.EndWithSuccess("Globalnet is disabled")
+	}
+
+	return true
+}
+
+func checkClusterGlobalEgressIps(cluster *cmd.Cluster, status *cli.Status) {
+	clusterGlobalEgress, err := cluster.SubmClient.SubmarinerV1().ClusterGlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining the clusterGlobalEgressIPs cr: %v", err))
+		return
+	}
+
+	if len(clusterGlobalEgress.Items) != 1 {
+		status.QueueFailureMessage(fmt.Sprintf("Number (%d) of clusterGlobalEgress resources != 1", len(clusterGlobalEgress.Items)))
+		return
+	}
+
+	if clusterGlobalEgress.Items[0].Name != constants.ClusterGlobalEgressIPName {
+		status.QueueFailureMessage(fmt.Sprintf("clusterGlobalEgress name (%s) not equals to default name", clusterGlobalEgress.Items[0].Name))
+		return
+	}
+}
+
+func checkGlobalEgressIps(cluster *cmd.Cluster, status *cli.Status) {
+	globalEgressIps, err := cluster.SubmClient.SubmarinerV1().GlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining GlobalEgressIPs resources: %v", err))
+		return
+	}
+
+	for i := range globalEgressIps.Items {
+		if *globalEgressIps.Items[i].Spec.NumberOfIPs != len(globalEgressIps.Items[i].Status.AllocatedIPs) {
+			status.QueueFailureMessage(fmt.Sprintf("Error GlobalEgressIPs(%s) NumberOfIPs != AllocatedIPs", globalEgressIps.Items[i].Name))
+			return
+		}
```

Comment:
The `NumberOfIPs` could be nil. Also check the status condition the same as I outlined above for the `ClusterGlobalEgressIP`.

```suggestion
                gip := globalEgressIps.Items[i]
                numberOfIPs := 1
                if gip.Spec.NumberOfIPs != nil {
                    numberOfIPs = *gip.Spec.NumberOfIPs
                }

		if numberOfIPs != len(gip.Status.AllocatedIPs) {
			status.Failure("The number of requested IPs (%d) does not match the number allocated (%d) for GlobalEgressIP %q",
                             numberOfIPs, len(gip.Status.AllocatedIPs), gip.Name)
		}
```